docs/chain/did.md

```diff
@@ -0,0 +1,73 @@
+
+A DID may be a "light" DID, which is not stored on-chain, or a "full" on-chain DID. A light DID is issued by default, with the keys stored locally on your device. By upgrading this to a full DID registered on the blockchain, all the keys associated with it can be retrieved from the KILT blockchain storage.
+
+A full DID can then be used to perform certain on-chain actions which include:
+
+* Writing CTypes to the chain
+* Writing attestations to the chain
+* Setting delegations
+* Doing key rotations on the DID keys
+
+## Registering a DID
+
+A full DID is needed if the user wants to become an attester or wants to setup delegations. A full DID also allows the user to embed a list of URLs, known as service endpoints, into the DID document so that it can be retrieved from the chain.
+
+To create a full DID the user first has to create a set of keys and service endpoints:
+
+* one authentication key for signing extrinsics from your DID
+* zero or more key agreement keys for encrypting messages that are sent to you
+* one attestation key for signing attestations (optional)
+* one delegation key for authorizing delegations (optional)
+* service endpoints that point to external hostings for others to find (optional)
+
+With these they are ready to write the DID to the KILT blockchain.
+
+The user then has to create the `did::create` extrinsic and sign it with any KILT account that has enough funding to pay both the transaction fees and the  DID deposit. The extrinsic consists of  
+
+* The `DidCreationDetails` object containing keys and service endpoints
+* The `DidSignature` which is a signature using your authentication key over the scale encoded `DidCreationDetails` from above
+* A regular signature authenticating the sender of the extrinsic
+
+Once this extrinsic is submitted and executed, the DID is written to the chain.
+
+## Updating a DID
+
+There is a whole set of extrinsics needed to update a full DID. These are:
+
+* `set_authentication_key`
+* `set_delegation_key`
+* `remove_delegation_key`
+* `set_attestation_key`
+* `remove_attestation_key`
+* `add_key_agreement_key`
+* `remove_key_agreement_key`
+* `add_service_endpoint`
+* `remove_service_endpoint`
+* `delete`
+
+
+All of them have to be authenticated using the DID that is updated. For that there is the special `submit_did_call` extrinsic. This extrinsic contains another call and a signature from the DID that is authorizing the inner call.
+
+For example when you want to add a new service endpoint:
+
+1. Create a `DidEndpoint` object containing a service id, a list of service types and a list of URLs.
+2. Wrap it in an `add_service_endpoint` extrinsic.
+3. Take this call and put it into a `DidAuthorizedCallOperation` object together with the DID, the tx_counter of the DID, the current block number and the account id of the submitter of the final extrinsic.
```

Comment:
I think the `tx_counter` was never explained? Also why the block number is included there is not explained too.